### PR TITLE
dbeaver/pro#1719 `application` connection property for Snowflake

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java [hunk 1]
```diff
@@ -70,6 +77,69 @@ protected boolean isPopulateClientAppName() {
...
+     *     LE = DBeaver_Lite <br/>
+     *     TE = DBeaver_Team <br/>
+     * <br/>
+     * CloudBeaver<br/>
+     *     EE = DBeaver_CloudBeaverEnterprise <br/>
+     *     TE = DBeaver_Team <br/>
+     *     CB AWS = DBeaver_CloudBeaverAWS <br/>
+     * <br/>
+     */
+    private String getPartnerApplicationIdentifier(String productName) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java]: Let's not use particular product identitfier here. 
Use DBUtils.getClientApplicationName
Just replace restricted characters with underscore

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java]: We need specific values that can be fixed

File: plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java [hunk 2]
```diff
@@ -70,6 +75,25 @@ protected boolean isPopulateClientAppName() {
...
+        @NotNull DBRProgressMonitor monitor,
+        JDBCExecutionContext context,
+        String purpose,
+        DBPConnectionConfiguration connectionInfo
+    ) throws DBCException {
+        Properties props = super.getAllConnectionProperties(monitor, context, purpose, connectionInfo);
+        String clientAppName = DBUtils.getClientApplicationName(container, context, null, false);
+        String appName = "DBeaver_";
+        if (clientAppName.contains("Team")) {
+            appName += "Team";
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.snowflake.core/src/org/jkiss/dbeaver/ext/snowflake/model/SnowflakeDataSource.java]: Let's not add special logic for TE

File: product/appstore/product-branding.properties [hunk 3]
```diff
@@ -4,7 +4,7 @@ appTitle=DBeaver
 productEclipseId=org.jkiss.dbeaver.core.appstore
 productModifier=ce
 productType=community
-productName=DBeaver CE
+productName=DBeaver Community
```
Comment (serge-rider) [inline product/appstore/product-branding.properties]: Wrong branding (it is for Windows Store)

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**